tests/test_config.py

```diff
@@ -387,6 +391,45 @@ def test_secure_mongo_advanced_options():
         config_object._Config__questions_mongo()
         assert not os.path.exists("/tmp/upsert_db_users")
 
+    # Try when setup has been already run once
+    # If it's an upgrade, users should not see:
+    # ╔══════════════════════════════════════════════════════╗
+    # ║ MongoDB root's and/or user's usernames have changed! ║
+    # ╚══════════════════════════════════════════════════════╝
+    config_object._Config__first_time = False
+    config_object._Config__config["mongo_secured"] = Config.FALSE
+
+    with patch("helpers.cli.CLI.colored_input") as mock_ci:
+        mock_ci.side_effect = iter([
+            "root",
+            "root_password",
+            "mongo_kobo_user",
+            "mongo_password"
+        ])
+        config_object._Config__questions_mongo()
+        assert os.path.exists("/tmp/upsert_db_users")
+        assert os.path.getsize("/tmp/upsert_db_users") == 0
+        os.remove("/tmp/upsert_db_users")
+
+    # Try when setup has been already run once
+    # If it's NOT an upgrade, Users should see:
+    # ╔══════════════════════════════════════════════════════╗
+    # ║ MongoDB root's and/or user's usernames have changed! ║
+    # ╚══════════════════════════════════════════════════════╝
+    config_object._Config__config["mongo_secured"] = Config.TRUE
+    with patch("helpers.cli.CLI.colored_input") as mock_ci:
+        mock_ci.side_effect = iter([
+            "root",
+            "root_passw0rd",
+            "kobo_user",
```

Comment:
Should this be: `...,"root_password", "mongo_kobo_user",...`?

Reply:
Nope, it's done on purpose to test password and username changes.